### PR TITLE
update migration-to-structured-logging.md

### PR DESCRIPTION
**Pull request by yangjunmyfm192085**
**Which issue(s) this PR fixes**:

Fixes  part of [#110737](https://github.com/kubernetes/kubernetes/issues/110737)

### CODE REVIEW

File: contributors/devel/sig-instrumentation/migration-to-structured-logging.md [hunk 1]
```diff
@@ -407,6 +416,7 @@ Here are a few exceptions to the rules above---some cases are temporary workarou
   then we should fallback to using object kind with suffix based on value type. For example `podName`, `podUID`.
 * When providing multiple indistinguishable values (for example list of evicted pods), then we can use plural version of
   argument name. For example we should use `pods` and not `podList`.
+* we can use `Klog.KObjSlice` for Kubernetes objects.
```
Comment (harshanarayana) [inline contributors/devel/sig-instrumentation/migration-to-structured-logging.md:419]: IMHO, this is required here. This section talks about `Here are a few exceptions to the rules above`.

File: contributors/devel/sig-instrumentation/migration-to-structured-logging.md [hunk 2]
```diff
@@ -467,16 +477,17 @@ func ChangePodStatus(newStatus, currentStatus string) {
 ## Good practice for passing values in structured logging
 
 When passing a value for a key-value pair, please use following rules:
-* Prefer using Kubernetes objects (for example `*v1.Pod`) and log them using `klog.KObj`
+* Prefer using Kubernetes objects (for example `*v1.Pod`) and log them using `klog.KObj` or `klog.KObjSlice`
```
Comment (harshanarayana) [inline contributors/devel/sig-instrumentation/migration-to-structured-logging.md:480]: `KObjSlice` is better off with resources having more than one entries. Using that for `*v1.Pod` for example is probably not a good idea right ?

File: contributors/devel/sig-instrumentation/migration-to-structured-logging.md [hunk 3]
```diff
@@ -467,16 +477,17 @@ func ChangePodStatus(newStatus, currentStatus string) {
...
   * When the original object is not available, use `klog.KRef` instead
+  * When object sets, we use `klog.KObjSlice`
```
Comment (harshanarayana) [inline contributors/devel/sig-instrumentation/migration-to-structured-logging.md:482]: I think this line needs a bit more clarification. Did you mean `When the object is slice type` ?

**Review by yangjunmyfm192085 — COMMENTED** [review]
@harshanarayana , Thanks for the very detailed review, I have applied all the suggestions, Could you please take a look again?

**Review by yangjunmyfm192085 — COMMENTED** [review]
/cc @logicalhan @serathius

**Review by pohly — APPROVED** [review]
/lgtm
/approve

**Pull request merged**